staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers.go

```diff
@@ -147,3 +150,34 @@ func IsStoredVersion(crd *CustomResourceDefinition, version string) bool {
 	}
 	return false
 }
+
+// GetCRDSubresourcesForVersion returns the subresources for given version in given CRD Spec.
+func GetCRDSubresourcesForVersion(crdSpec *CustomResourceDefinitionSpec, version string) (*CustomResourceSubresources, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceWebhookConversion) {
```

Comment:
helpers are not the right place for feature gate dependent code. This logic belongs into the implementation of subresources. Having the gates spread everywhere make reasoning about the code impossible and any change error prone.

Reply:
Or in other words: apis packages should not depend on features gate, the storage registry including strategy should.

Reply:
@sttts how would you apply your views to CRD-based types where there is no way to put code in the storage registry/strategy?

Reply:
> You can't use them to change semantics of features, only whether people are allowed to use the feature.

This is not my point. We could have the gate her in the accessors to the API. My point is that we should not spread them all over the code base as it makes reasoning about the behaviour impossible. In other words: constraint the accessors with the gate, but leave the registry logic generic, or the other way around. But this mixture is an anti-pattern.

Reply:
> @sttts how would you apply your views to CRD-based types where there is no way to put code in the storage registry/strategy?

I think the perspective here is that the feature gate usage doesn't belong in an api package.  Rather than add a helper function inside of the api package, these can live in the package where they are used.  It is consistent with the idea that the package shouldn't hold functions.  This file only exists because I followed a pattern I have since regretted in openshift.

I suspect that the original comment from @sttts would be satisfied by moving them to the calling package and making them private to keep the logic contained.